46.md

```diff
@@ -45,15 +45,25 @@ _remote-signer_ provides connection token in the form:
 bunker://<remote-signer-pubkey>?relay=<wss://relay-to-connect-on>&relay=<wss://another-relay-to-connect-on>&secret=<optional-secret-value>
 ```
 
-_user_ passes this token to _client_, which then sends `connect` request to _remote-signer_ via the specified relays. Optional secret can be used for single successfully established connection only, _remote-signer_ SHOULD ignore new attempts to establish connection with old secret. 
+_user_ passes this token to _client_, which then sends `connect` request to _remote-signer_ via the specified relays. Optional secret can be used for single successfully established connection only, _remote-signer_ SHOULD ignore new attempts to establish connection with old secret.
 
 ### Direct connection initiated by the _client_
 
-_client_ provides a connection token in the form: 
+_client_ provides a connection token using `nostrconnect://` as the protocol, and `client-pubkey` as the origin. Additional information should be passed as query parameters:
+
+- `relay` (required) - one or more relay urls on which the _client_ is listening for responses from the _remote-signer_.
+- `secret` (required) - a short random string that the _remote-signer_ should return as the `result` field of its response.
```

Comment:
I'm not sure that this is correct.

Per the details below, when the remote signer receives the `nostrconnect://` token, it is expected to send a `connect` request to the client.  This `connect` request should include the secret that was contained in the `nostrconnect://` token.

So the entry for `secret` should read something like this:

```suggestion
- `secret` (required) - a short random string that the _remote-signer_ should include as a param when it sends its `connect` request to the client.
```

Reply:
A connect request from bunker to client is undefined, so the proposed change doesn't make sense. I'll add some clarification.

Reply:
Actually, it's already in the spec:

> _user_ passes this token to _remote-signer_, which then sends `connect` *response* event to the `client-pubkey` via the specified relays. Client discovers `remote-signer-pubkey` from connect response author. `secret` value MUST be provided to avoid connection spoofing, _client_ MUST validate the `secret` returned by `connect` response.

Reply:
Ah I think I see now.  The `nostrconnect://` string is treated as the `connect` request, in this case, and the client listens for a `connect` _response_ as if it had actually sent a `connect` _request_.